.github/workflows/run_tests.yml

```diff
@@ -144,54 +147,57 @@ jobs:
 
+      # Begin running the tests using various DB deployments
       - name: Single-shard DB tests
         run: |
-          SCRATCH_DIR="single_scratch"
+          SCRATCH_DIR="single_scratch" && PORT=6380
           rm -rf $SCRATCH_DIR; mkdir $SCRATCH_DIR; pushd $SCRATCH_DIR &&
-          redis-server --port 6380 --daemonize yes \
+          redis-server --port $PORT --daemonize yes \
                        --logfile "single.log" \
                        --loadmodule $REDISAI_MODULES &&
-          sleep 5; cat single.log &&
           popd &&
           export SSDB=$SSDB_SINGLE SMARTREDIS_TEST_CLUSTER=False &&
+          utils/check_redis.sh $PORT &&
           make test-verbose COV_FLAGS="--cov=./src/python/module/smartredis/ --cov-report=xml --cov-append"
 
       - name: Clustered DB tests
         run: |
-          SCRATCH_DIR="cluster_scratch"
+          SCRATCH_DIR="cluster_scratch" &&
           rm -rf $SCRATCH_DIR; mkdir $SCRATCH_DIR; pushd $SCRATCH_DIR &&
           for port in {7000..7002}; do
             redis-server --port $port --cluster-enabled yes --daemonize yes \
                          --cluster-config-file "$port.conf" --protected-mode no --save "" \
                          --logfile "$port.log" \
                          --loadmodule $REDISAI_MODULES
           done &&
-          sleep 2 &&
+          popd &&
+          for port in {7000..7002}; do ../utils/check_redis.sh $port done &&
           redis-cli --cluster create $(echo $SSDB_CLUSTERED | tr "," " ") --cluster-yes &&
           export SSDB=$SSDB_CLUSTERED SMARTREDIS_TEST_CLUSTER=True &&
-          popd &&
           python -m pytest --cov=./src/python/module/smartredis/ --cov-report=xml --cov-append \
                  --ignore ./tests/docker -vv -s ./tests
 
       - name: UDS DB tests
         run: |
+          SOCKET="/tmp/redis.sock" &&
           mkdir -p /tmp && ## Create socketfile for UDS testing
-          touch /tmp/redis.sock &&
-          chmod 777 /tmp/redis.sock &&
+          touch $SOCKET &&
+          chmod 777 $SOCKET &&
           SCRATCH_DIR="uds_scratch" &&
           rm -rf $SCRATCH_DIR; mkdir $SCRATCH_DIR; pushd $SCRATCH_DIR &&
-          redis-server --unixsocket /tmp/redis.sock --unixsocketperm 777 --port 0 --bind 127.0.0.1 \
+          redis-server --unixsocket $SOCKET --unixsocketperm 777 --port 0 --bind 127.0.0.1 \
                        --daemonize yes --protected-mode no --logfile "uds.log" \
                        --loadmodule $REDISAI_MODULES &&
           popd &&
+          utils/check_redis.sh $SOCKET
           export SSDB=$SSDB_UDS SMARTREDIS_TEST_CLUSTER=False &&
           python -m pytest --cov=./src/python/module/smartredis/ --cov-report=xml --cov-append \
                  --ignore ./tests/docker -vv -s ./tests
 
-      - name: Collect C++ coverage
-        run: third-party/lcov/install/usr/local/bin/lcov -c -d build/CMakeFiles -d tests -o coverage.info
+      # Process and upload code coverage
```

Comment:
Could you add a comment that the Python coverage was collected in a previous step?

Reply:
For sure